nativelib/src/main/scala/scala/scalanative/native/signal.scala

```diff
@@ -1,34 +1,374 @@
+
+  @name("scalanative_libc_ill_illopc")
+  def ILL_ILLOPC: CInt = extern
+
+  @name("scalanative_libc_ill_illopn")
+  def ILL_ILLOPN: CInt = extern
+
+  @name("scalanative_libc_ill_illadr")
+  def ILL_ILLADR: CInt = extern
+
+  @name("scalanative_libc_ill_illtrp")
+  def ILL_ILLTRP: CInt = extern
+
+  @name("scalanative_libc_ill_prvopc")
+  def ILL_PRVOPC: CInt = extern
+
+  @name("scalanative_libc_ill_prvreg")
+  def ILL_PRVREG: CInt = extern
+
+  @name("scalanative_libc_ill_coproc")
+  def ILL_COPROC: CInt = extern
+
+  @name("scalanative_libc_ill_badstk")
+  def ILL_BADSTK: CInt = extern
+
+  @name("scalanative_libc_fpe_intdiv")
+  def FPE_INTDIV: CInt = extern
+
+  @name("scalanative_libc_fpe_intovf")
+  def FPE_INTOVF: CInt = extern
+
+  @name("scalanative_libc_fpe_fltdiv")
+  def FPE_FLTDIV: CInt = extern
+
+  @name("scalanative_libc_fpe_fltovf")
+  def FPE_FLTOVF: CInt = extern
+
+  @name("scalanative_libc_fpe_fltund")
+  def FPE_FLTUND: CInt = extern
+
+  @name("scalanative_libc_fpe_fltres")
+  def FPE_FLTRES: CInt = extern
+
+  @name("scalanative_libc_fpe_fltinv")
+  def FPE_FLTINV: CInt = extern
+
+  @name("scalanative_libc_fpe_fltsub")
+  def FPE_FLTSUB: CInt = extern
+
+  @name("scalanative_libc_segv_maperr")
+  def SEGV_MAPERR: CInt = extern
+
+  @name("scalanative_libc_segv_accerr")
+  def SEGV_ACCERR: CInt = extern
+
+  @name("scalanative_libc_bus_adraln")
+  def BUS_ADRALN: CInt = extern
+
+  @name("scalanative_libc_bus_adrerr")
+  def BUS_ADRERR: CInt = extern
+
+  @name("scalanative_libc_bus_objerr")
+  def BUS_OBJERR: CInt = extern
+
+  @name("scalanative_libc_trap_brkpt")
+  def TRAP_BRKPT: CInt = extern
+
+  @name("scalanative_libc_trap_trace")
+  def TRAP_TRACE: CInt = extern
+
+  @name("scalanative_libc_cld_exited")
+  def CLD_EXITED: CInt = extern
+
+  @name("scalanative_libc_cld_killed")
+  def CLD_KILLED: CInt = extern
+
+  @name("scalanative_libc_cld_dumped")
+  def CLD_DUMPED: CInt = extern
+
+  @name("scalanative_libc_cld_trapped")
+  def CLD_TRAPPED: CInt = extern
+
+  @name("scalanative_libc_cld_stopped")
+  def CLD_STOPPED: CInt = extern
+
+  @name("scalanative_libc_cld_continued")
+  def CLD_CONTINUED: CInt = extern
+
+  @name("scalanative_libc_poll_in")
+  def POLL_IN: CInt = extern
+
+  @name("scalanative_libc_poll_out")
+  def POLL_OUT: CInt = extern
+
+  @name("scalanative_libc_poll_msg")
+  def POLL_MSG: CInt = extern
+
+  @name("scalanative_libc_poll_err")
+  def POLL_ERR: CInt = extern
+
+  @name("scalanative_libc_poll_pri")
+  def POLL_PRI: CInt = extern
+
+  @name("scalanative_libc_poll_hup")
+  def POLL_HUP: CInt = extern
+
+  @name("scalanative_libc_si_user")
+  def SI_USER: CInt = extern
+
+  @name("scalanative_libc_si_queue")
+  def SI_QUEUE: CInt = extern
+
+  @name("scalanative_libc_si_timer")
+  def SI_TIMER: CInt = extern
+
+  @name("scalanative_libc_si_asyncio")
+  def SI_ASYNCIO: CInt = extern
+
+  @name("scalanative_libc_si_mesgq")
+  def SI_MESGQ: CInt = extern
+}
+
+object signalH {
```

Comment:
This second object is mandatory due to the recursive type ucontext_t, to handle it we need to create the implicit class and since it contains function definitions it cannot be in the main object (@extern prohibit function definition)